mozilla-ai/lumigator
# Separate dev dependencies and add pre-commit support

## What's changing

I realized that we were introducing dev dependencies to the image, like pytest. At the same time, I also discover that we weren't installing anywhere the pre-commit dependency, and we didn't configure it. 

So, I fix it. I modified the compose and the Dockerfile and the compose, and I created a new uv/python project in the root of the repo only including the pre-commit. When we execute `make local-up`, it also configures the pre-commit.

As long as the pre-commit is configured at a repo level inside .git/webhooks/, after execute the `make local-up` you can go to whatever folder and follow the development instructions `uv sync`, etc. 

## How to test it

Clean the .venv folders if you already have one (or more), just to make sure, and execute `make local-up`, go to the backend folder, run `uv sync,` and start developing! 

## Additional notes for reviewers

## I already...
